### PR TITLE
Validations, Model Specs, and n+1 Problems

### PR DESCRIPTION
**Pull request by luftedar**
### Hello :wave: 
# Feature: Validations, Model Specs, and n+1 Problems

In this update: 

- Added the following validations:
  - For the `User` model:
     - `Name` must not be blank.
     - `PostsCounter` must be an integer greater than or equal to zero.
   - For the `Post` model:
      - Title must not be `blank`.
      - Title must not exceed `250 characters`.
      - `CommentsCounter` must be an integer greater than or equal to zero.
      - `LikesCounter` must be an integer greater than or equal to zero.
- Added `unit specs` for all of your models' methods and validations.
- Added `flash messages` in the create actions in all your controllers.

Have a nice CR. :+1:

### CODE REVIEW

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+    like.build_user(name: 'Orçun')
+    like.build_post(text: '')
+    expect(like).to be_valid
+  end
+
+  it 'updates a posts likes correctly' do
+    user = User.new(name: 'Tom', posts_counter: 0)
+    user2 = User.create!(name: 'Lilly', posts_counter: 0)
+    post = user2.posts.create!(title: 'Post1', text: 'This is a post', likes_counter: 0, comments_counter: 0)
+    post.likes.create!(user:)
```
Comment (Dipeshtwis) [inline spec/models/like_spec.rb:15]: - [ ] It gives me syntax errors when I'm running `rspec`. I think this line should be `post.likes.create!(user: user)`. Please fix that.

Screenshot from 2022-03-05 01-39-36

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @luftedar, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

- [ ] I would recommend including all the steps to run and test the project.

```
# to setup and migrate the database
rails db:setup

# To test the application
rspec
```
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N.A.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by luftedar** [comment]
### Hello :wave: 
# Please note here before reviewing the project.
[Changes required by the previous code reviewer ](https://github.com/luftedar/blog-app/pull/8#discussion_r819881383). Already works well on my side. Please check the following:

1- When I run `rspec` it works well:



2- If I change the code as requested. It still works:



3- When I run `rubocop` it gives me warning:



4- So auto fix changes line:

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi @luftedar🥇,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**